src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -266,6 +284,71 @@ static void checkAccessRightsForSelect(
     context->checkAccess(AccessType::SELECT, table_id, syntax_analyzer_result.requiredSourceColumnsForAccessCheck());
 }
 
+/// Parse additional filter for table in format 'table.name:expression'
+/// Examples: 'default.hits:UserID=12345', 'visits:UserID = 0 ? 1 : 0'
+static ASTPtr parseAdditionalFilterConditionForTable(
+    const char * start,
+    const char * end,
+    const DatabaseAndTableWithAlias & target,
+    const Context & context)
+{
+    const char delimiter = ':';
+
+    const char * pos = start;
+    for (; pos < end; ++pos)
+        if (*pos == delimiter)
```

Comment:
This is wrong, the column name can contain `:`

Reply:
it is ok - we always skip only till first `:`.
But if table name can contain `:` then there is a trouble